File: api/app/models/recommendation.py
from enum import Enum as PyEnum

from beanie import Document

from .base import TimestampMixin


class RecommendationType(str, PyEnum):
    POSITIVE = "POSITIVE"
    WARNING = "WARNING"
    ACTION_REQUIRED = "ACTION_REQUIRED"
    INFO = "INFO"


class Priority(str, PyEnum):
    HIGH = "HIGH"
    MEDIUM = "MEDIUM"
    LOW = "LOW"


class Recommendation(Document, TimestampMixin):
    id: str  # type: ignore[assignment]
    hive_id: str
    type: RecommendationType
    title: str
    description: str
    priority: Priority = Priority.MEDIUM

    class Settings:
        name = "recommendations"
        indexes = ["hive_id"]

File: composeApp/src/commonMain/kotlin/com/beekeeper/app/domain/model/Recommendation.kt
package com.beekeeper.app.domain.model

import kotlinx.serialization.Serializable

@Serializable
data class Recommendation(
    val id: String,
    val hiveId: String,
    val type: RecommendationType,
    val title: String,
    val description: String,
    val priority: Priority
)

enum class RecommendationType {
    POSITIVE,
    WARNING,
    ACTION_REQUIRED,
    INFO
}

enum class Priority {
    HIGH,
    MEDIUM,
    LOW
}
